extension/experiments/voice/api.js

```diff
@@ -101,6 +733,28 @@ this.voice = class extends ExtensionAPI {
           async browserOpenAddonsMgr() {
             return runCommand("Tools:Addons");
           },
+
+          async sendKeyboardEvent(tabId, eventName, eventProperties) {
+            function getTabOrActive(tabId) {
+              const tab =
+                tabId !== null
+                  ? tabTracker.getTab(tabId)
+                  : tabTracker.activeTab;
+              if (!context.canAccessWindow(tab.ownerGlobal)) {
+                throw new ExtensionError(
+                  tabId === null
+                    ? "Cannot access activeTab"
+                    : `Invalid tab ID: ${tabId}`
+                );
+              }
+              return tab;
+            }
+
+            const tab = getTabOrActive(tabId);
+            console.log("tab is", tab, tab.KeyboardEvent);
+            const KeyboardEvent = new KeyboardEvent(eventName, eventProperties);
```

Comment:
For eventProperties see https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/KeyboardEvent